bin/pipeline_result_processor/prp/cli.py

```diff
@@ -0,0 +1,155 @@
+import json
+import logging
+
+import click
+from pydantic import ValidationError
+
+from .models.metadata import RunInformation, SoupVersion
+from .models.phenotype import PhenotypeType
+from .models.sample import MethodIndex, PipelineResult
+from .models.qc import QcMethodIndex
+from .parse import (
+    parse_cgmlst_results,
+    parse_mlst_results,
+    parse_qust_results,
+    parse_resistance_pred,
+    parse_species_pred,
+    parse_virulence_pred,
+)
+
+logging.basicConfig(
+    level=logging.INFO, format="[%(asctime)s] %(levelname)s in %(module)s: %(message)s"
+)
+LOG = logging.getLogger(__name__)
+
+OUTPUT_SCHEMA_VERSION = 1
+
+
+@click.group()
+def cli():
+    pass
+
+
+@cli.command()
+@click.option("-i", "--sample-id", required=True, help="Sample identifier")
+@click.option(
+    "-u",
+    "--run-metadata",
+    type=click.File(),
+    required=True,
+    help="Analysis metadata from the pipeline in json format",
+)
+@click.option("-q", "--quast", type=click.File(), help="Quast quality control metrics")
+@click.option(
+    "-p",
+    "--process-metadata",
+    type=click.File(),
+    multiple=True,
+    help="Nextflow processes metadata from the pipeline in json format",
+)
+@click.option(
+    "-k", "--kraken", type=click.File(), help="Kraken specie annotation results"
+)
+@click.option("-m", "--mlst", type=click.File(), help="MLST prediction results")
+@click.option("-c", "--cgmlst", type=click.File(), help="cgMLST prediction results")
+@click.option(
+    "-v", "--virulence", type=click.File(), help="Virulence factor prediction results"
+)
+@click.option(
+    "-r", "--resistance", type=click.File(), help="Resistance prediction results"
+)
+@click.argument("output", type=click.File("w"))
+def create_output(
+    sample_id,
+    run_metadata,
+    quast,
+    process_metadata,
+    kraken,
+    mlst,
+    cgmlst,
+    virulence,
+    resistance,
+    output,
```

Comment:
Im not sure if blacks adherence to "one word per row when splitting" looks great with function headers